spec/ddtrace/contrib/rest_client/request_patch_spec.rb

```diff
@@ -0,0 +1,170 @@
+require 'spec_helper'
+require 'ddtrace'
+require 'ddtrace/contrib/rest_client/request_patch'
+require 'rest_client'
+
+RSpec.describe Datadog::Contrib::RestClient::RequestPatch do
+  let(:tracer) { Datadog::Tracer.new(writer: FauxWriter.new) }
+  let(:rest_client_options) { {} }
+
+  before do
+    Datadog.configure do |c|
+      c.use :rest_client, rest_client_options.merge(tracer: tracer)
+    end
+
+    WebMock.disable_net_connect!
+    WebMock.enable!
```

Comment:
Is WebMock necessary? Surely we don't want it to make any outgoing HTTP requests, but is there some mock we can put in at the RSpec level to negate the need to depend on this? If not, it's okay, and we can keep it: just wondering if there was a way of keeping this simpler.

Reply:
We would have to mock quite a few internals in `rest_client` to do so.
Webmock allows us to treat those as a black box, and write more integration style tests.

Reply:
Hmmm, I prefer unit tests where we're clear about the inputs/outputs (makes it easier to break things apart), however, I can see the value of an integration test here. This is probably okay.